src/defines/defines-creator.ts

```diff
@@ -0,0 +1,52 @@
+import {IDefines} from './defines';
+import {IDefine} from './define';
+import {MagickFormat} from '../magick-format';
+import {MagickDefine} from './magick-define';
+
+interface IHasToString {
+  toString(): string
+}
+
+export abstract class DefinesCreator implements IDefines {
+  format: MagickFormat;
+
+  constructor (format: MagickFormat) {
+    this.format = format;
+  }
+
+  createDefine(name: string, value: boolean): MagickDefine;
+  createDefine(name: string, value: number): MagickDefine;
+  createDefine(name: string, value: string): MagickDefine;
+  createDefine(name: string, value: unknown[]): MagickDefine | null;
```

Comment:
I would prefer it if we would only allow `boolean`, `number` and `string` for now.

Reply:
I removed overload for arrays